language-service/src/parser/jsonParser.ts

```diff
@@ -709,14 +714,53 @@ export class ObjectASTNode extends ASTNode {
 				seenKeys[key] = node.value;
 				unprocessedProperties.push(key);
 			}
-			
 		});
 
+		let findMatchingProperties = (propertyKey: string): ASTNodeMap => {
+			let result: ASTNodeMap = Object.create(null);
+			const compareKey: string = propertyKey.toUpperCase();
+
+			Object.keys(seenKeys).forEach((propertyName: string) => {
+				if (propertyName.toUpperCase() === compareKey) {
+					result[propertyName] = seenKeys[propertyName];
+				}
+			});
+
+			return result;
+		}
+
+		let findSeenKeys = (propertyKey: string): ASTNodeMap => {
+			if (schema.properties) {
+				const propSchema: JSONSchema = schema.properties[propertyKey];
+				const ignoreKeyCase: boolean = this.getIgnoreKeyCase(propSchema);
+
+				if (ignoreKeyCase) {
+					return findMatchingProperties(propertyKey);
+				}
+			}
+
+			let result: ASTNodeMap = Object.create(null);
+			if (seenKeys[propertyKey]) {
```

Comment:
This will never be true as you already checked this in hasProperty function. maybe combine these 2 functions in one?